test/unit/org/apache/cassandra/io/util/Files.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.io.util;
+
+import java.io.IOException;
+import java.nio.file.FileSystem;
+
+import com.google.common.base.StandardSystemProperty;
+import com.google.common.jimfs.Configuration;
+import com.google.common.jimfs.Jimfs;
+
+import org.apache.cassandra.io.filesystem.ForwardingFileSystem;
+import org.apache.cassandra.io.filesystem.ListenableFileSystem;
+
+public class Files
+{
+    public static ListenableFileSystem newInMemoryFileSystem()
+    {
+        return new ListenableFileSystem(jimfs());
+    }
+
+    public static ListenableFileSystem newGlobalInMemoryFileSystem()
+    {
+        return global(jimfs());
+    }
+
+    public static ListenableFileSystem global()
+    {
+        return global(File.unsafeGetFilesystem());
+    }
+
+    public static ListenableFileSystem global(FileSystem real)
+    {
+        FileSystem current = File.unsafeGetFilesystem();
+        ListenableFileSystem fs = new ListenableFileSystem(new ForwardingFileSystem(real)
+        {
+            @Override
+            public void close() throws IOException
```

Comment:
nit: Does anything ever actually close this?

Reply:
if you try to create a new FS per test you need to, else the old fs can't free memory... jimfs registers itself with the global `FileSystems`, so a reference will be held if you do not *close*

Reply:
in the single example in this PR, its localized to the class... so don't actually need to... but I wanted to make sure you can create a fs for your test even if another exists, or if each test wants to create in isolation...